File: study-design-patterns/ts/factory-pattern/product/FileHandler.ts
import IOHandler from "./IOHandler";

export default class FileHandler extends IOHandler {
  add(): void {
    console.log('FileHandler add method call!')
  }

  query(): void {
    console.log('FileHandler query method call!')
  }

  remove(): void {
    console.log('FileHandler remove method call!')
  }

  update(): void {
    console.log('FileHandler update method call!')
  }
}

File: study-design-patterns/java/factory-pattern/src/product/FileHandler.java
package product;

public class FileHandler extends IOHandler{
    @Override
    public void add(String id, String name) {
        System.out.print("FileHandler add method call!\n");
    }

    @Override
    public void remove(String id) {
        System.out.print("FileHandler remove method call!\n");
    }

    @Override
    public void update(String id, String name) {
        System.out.print("FileHandler update method call!\n");
    }

    @Override
    public void query(String id) {
        System.out.print("FileHandler query method call!\n");
    }
}
